oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -482,17 +481,52 @@ private ImpersonatedCredentials(Builder builder) {
     }
   }
 
+  /**
+   * Gets the universe domain for the credential.
+   *
+   * @return An explicit universe domain if it was explicitly provided, use the universe domain from
+   *     source credentials otherwise
+   */
+  @Override
+  public String getUniverseDomain() throws IOException {
+    if (isExplicitUniverseDomain()) {
+      return super.getUniverseDomain();
+    }
+    return this.sourceCredentials.getUniverseDomain();
+  }
+
+  /**
+   * Checks if universe domain equals to {@link Credentials#GOOGLE_DEFAULT_UNIVERSE}.
+   *
+   * @return true if universeDomain equals to {@link Credentials#GOOGLE_DEFAULT_UNIVERSE}, false
+   *     otherwise
+   */
+  @Override
+  boolean isDefaultUniverseDomain() {
```

Comment:
We should probably change the super class's [implementation](https://github.com/googleapis/google-auth-library-java/blob/89201558db913d9a71b3acccbab8eb0045ada6de/oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java#L270) to what you have here.
From 
```
this.universeDomain.equals(Credentials.GOOGLE_DEFAULT_UNIVERSE)
```
to
```
getUniverseDomain().equals(Credentials.GOOGLE_DEFAULT_UNIVERSE)
```
so that we don't have to duplicate the effort here. Make sure it does not break other scenarios though.

Reply:
Good point! Let me try it out.

Reply:
I checked and this should not conflict with other use cases, this `isDefaultUniverseDomain()` is only used here and from ServiceAccountCredentials. In fact, I think use getUniverseDomain() inside it should be the expected behavior, otherwise it won't accurately reflect status for subclasses. e.g. ComputeEngineCredentials which also have [custom logic](https://github.com/googleapis/google-auth-library-java/blob/89201558db913d9a71b3acccbab8eb0045ada6de/oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java#L224) for getting universe domain.

I am a bit hesitant on how to deal with the exception situation here. @blakeli0 Do you have any suggestions?

In parent class GoogleCredentials,
```
 // throwing exception in isDefaultUniverseDomain would be breakiing change
  boolean isDefaultUniverseDomain() {
    try {
     // getUniverseDomain() throws exception
      return getUniverseDomain().equals(Credentials.GOOGLE_DEFAULT_UNIVERSE);
    } catch (IOException ignored) {
         logger.warn("Error getting universe domain: " + e.getMessage(), e);
         return false; // this might be misleading
    }
  }
```

Reply:
Throwing exception in `isDefaultUniverseDomain` is not a breaking change because it is package private. I think it should be fine.

Reply:
ah- somehow missed this. Thank you.